lib/iaas/GcpClient.js

```diff
@@ -0,0 +1,200 @@
+      })
+      .catch(BaseCloudClient.providerErrorTypes.Forbidden, err => {
+        logger.error(err.message);
+        throw new Forbidden(`Authentication at google cloud storage provider failed while deleting blob ${file} in container ${container}`);
+      })
+      .catchThrow(BaseCloudClient.providerErrorTypes.NotFound,
+        new NotFound(`Object '${file}' not found while deleting in container ${container}`));
+  }
+
+  download(options) {
+    return utils.streamToPromise(this.storageClient
+        .bucket(options.container)
+        .file(options.remote)
+        .createReadStream())
+      .catchThrow(BaseCloudClient.providerErrorTypes.NotFound, new NotFound(`Object '${options.remote}' not found`));
+  }
+
+  upload(options, buffer) {
+    return new Promise((resolve, reject) => {
+      function cleanup() {
+        uploadedStream.removeListener('finish', onfinish);
+        uploadedStream.removeListener('error', onerror);
+      }
+
+      function onerror(err) {
+        cleanup();
+        reject(err);
+      }
+
+      function onfinish() {
+        cleanup();
+        resolve(JSON.parse(buffer));
+      }
+
+      let uploadedStream = this.storageClient
+        .bucket(options.container)
+        .file(options.remote)
+        .createWriteStream();
+
+      uploadedStream.on('error', onerror);
+      uploadedStream.on('finish', onfinish);
+      uploadedStream.end(buffer);
+    });
+  }
+
+  uploadJson(container, file, data) {
+    if (arguments.length < 3) {
+      data = file;
+      file = container;
+      container = this.containerName;
+    }
+    return this
+      .upload({
+        container: container,
+        remote: file
+      }, new Buffer(JSON.stringify(data, null, 2), 'utf8'));
+  }
+
+  downloadJson(container, file) {
+    if (arguments.length < 2) {
+      file = container;
+      container = this.containerName;
+    }
+    return this
+      .download({
+        container: container,
+        remote: file
+      })
+      .then((data) => JSON.parse(data))
+      .catchThrow(SyntaxError, new NotFound(`Object '${file}' not found`));
+  }
+
+  deleteSnapshot(snapshotName) {
+    return Promise
+      .try(() => {
+        return this.computeClient
+          .snapshot(snapshotName)
+          .delete()
+          .then(retval => logger.info(`Deleted snapshot ${snapshotName}`, retval));
+      })
+      .catch(BaseCloudClient.providerErrorTypes.Unauthorized, err => {
+        logger.error(err.message);
+        throw new Unauthorized(`Authorization at google compute failed while deleting snapshot ${snapshotName}`);
+      })
+      .catch(BaseCloudClient.providerErrorTypes.Forbidden, err => {
+        logger.error(err.message);
+        throw new Forbidden(`Authentication at google compute failed while deleting snapshot ${snapshotName}`);
+      })
+      .catchThrow(BaseCloudClient.providerErrorTypes.NotFound,
```

Comment:
same comments as that provided in `remove` method.

Reply:
Replaced with catchThrow().